src/lsdb/catalog/catalog.py

```diff
@@ -221,20 +221,22 @@ def cone_search(self, ra: float, dec: float, radius: float):
         ddf_partition_map = {pixel: i for i, pixel in enumerate(pixels_in_cone)}
         return Catalog(cone_search_ddf, ddf_partition_map, filtered_hc_structure)
 
-    def polygon_search(self, polygon: SingleSphericalPolygon) -> Catalog:
+    def polygon_search(self, vertices: List[SphericalCoordinates]) -> Catalog:
         """Perform a polygonal search to filter the catalog.
 
         Filters to points within the polygonal region specified in ra and dec, in degrees.
         Filters partitions in the catalog to those that have some overlap with the region.
 
         Args:
-            polygon (SingleSphericalPolygon): The polygon to filter pixels with.
+            vertices (List[SphericalCoordinates]): The list of vertices of the polygon to
```

Comment:
I think the type alias in the docstring isn't as helpful as writing out the full tuple type. It's nice in the code, but I think for a user there's no way of finding out what the type should be without digging into the code and inspecting the definition in hipscat.

Reply:
I agree, from the users' perspective having the explicit type makes their life easier. I'm changing it!